identityClient/identityClient_test.go

```diff
@@ -3476,3 +3478,58 @@ func TestListCreatedAccessRequestPolicy(t *testing.T) {
 	}
 
 }
+
+func TestRealmSettingsUpdateRequesMarshalUnmarshalCorrectly(t *testing.T) {
+	secretsEnabled := true
+	mfaAvailable := []string{"one", "two"}
+	emailLookupsEnabled := true
+	tozIDFederationEnabled := true
+	mpcEnabled := true
+	forgotPasswordCustomLink := "http://custom-link.com"
+	forgotPasswordCustomText := "Call helpdesk"
+
+	realmUpdateSettingsRequest := RealmSettingsUpdateRequest{
+		SecretsEnabled:           &secretsEnabled,
+		MFAAvailable:             &mfaAvailable,
+		EmailLookupsEnabled:      &emailLookupsEnabled,
+		TozIDFederationEnabled:   &tozIDFederationEnabled,
+		MPCEnabled:               &mpcEnabled,
+		ForgotPasswordCustomLink: &forgotPasswordCustomLink,
+		ForgotPasswordCustomText: &forgotPasswordCustomText,
+	}
+
+	marshalled, err := json.Marshal(realmUpdateSettingsRequest)
+	if err != nil {
+		t.Fatalf("can't marshal RealmSettingsUpdateRequest %s", err.Error())
+
+	}
+
+	var unmarshalledRealmUpdateSettingsRequest RealmSettingsUpdateRequest
+	err = json.Unmarshal(marshalled, &unmarshalledRealmUpdateSettingsRequest)
+	if err != nil {
+		t.Fatalf("can't unmarshal RealmSettingsUpdateRequest %s", err.Error())
+
+	}
+	verifyFieldUnmarshalledCorrectly(t, "SecretsEnabled", *realmUpdateSettingsRequest.SecretsEnabled,
+		*unmarshalledRealmUpdateSettingsRequest.SecretsEnabled)
+	verifyFieldUnmarshalledCorrectly(t, "EmailLookupsEnabled", *realmUpdateSettingsRequest.EmailLookupsEnabled,
+		*unmarshalledRealmUpdateSettingsRequest.EmailLookupsEnabled)
+	verifyFieldUnmarshalledCorrectly(t, "TozIDFederationEnabled", *realmUpdateSettingsRequest.TozIDFederationEnabled,
+		*unmarshalledRealmUpdateSettingsRequest.TozIDFederationEnabled)
+	verifyFieldUnmarshalledCorrectly(t, "MPCEnabled", *realmUpdateSettingsRequest.MPCEnabled,
+		*unmarshalledRealmUpdateSettingsRequest.MPCEnabled)
+	verifyFieldUnmarshalledCorrectly(t, "ForgotPasswordCustomLink", *realmUpdateSettingsRequest.ForgotPasswordCustomLink,
+		*unmarshalledRealmUpdateSettingsRequest.ForgotPasswordCustomLink)
+	verifyFieldUnmarshalledCorrectly(t, "ForgotPasswordCustomText", *realmUpdateSettingsRequest.ForgotPasswordCustomText,
+		*unmarshalledRealmUpdateSettingsRequest.ForgotPasswordCustomText)
+	if !reflect.DeepEqual(*realmUpdateSettingsRequest.MFAAvailable, *unmarshalledRealmUpdateSettingsRequest.MFAAvailable) {
+		t.Fatalf("MFAAvailable has unmarshalled incorectly, expected: %s, actual: %s", *realmUpdateSettingsRequest.MFAAvailable,
+			*unmarshalledRealmUpdateSettingsRequest.MFAAvailable)
+	}
+}
+
+func verifyFieldUnmarshalledCorrectly(t *testing.T, fieldName string, expected interface{}, actual interface{}) {
```

Comment:
Nice! This would be a good candidate function for moving here to allow it to be used by other test files in other service repositories -> https://github.com/tozny/utils-go/blob/master/test/test.go

or moving here https://github.com/tozny/e3db-clients-go/blob/master/test/test.go so that its available to use for any other client tests in this repo

Reply:
Thanks! I wanted to find something like that. Let me move it there.